osu.Game/Screens/Play/HUD/BPMCounter.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Localisation;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Skinning;
+
+namespace osu.Game.Screens.Play.HUD
+{
+    public partial class BPMCounter : RollingCounter<double>, ISkinnableDrawable
+    {
+        protected override double RollingDuration => 750;
+
+        [Resolved]
+        private IBindable<WorkingBeatmap> beatmap { get; set; } = null!;
+
+        [Resolved]
+        protected IGameplayClock GameplayClock { get; private set; } = null!;
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colour)
+        {
+            Colour = colour.BlueLighter;
+            Current.Value = DisplayedCount = 0;
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            //We dont want it going to 0 when we pause. so we block the updates
+            if (GameplayClock.IsPaused.Value) return;
+
+            // We want to check Rate every update to cover windup/down
+            Current.Value = beatmap.Value.Beatmap.ControlPointInfo.TimingPointAt(GameplayClock.CurrentTime).BPM * GameplayClock.Rate;
+        }
+
+        protected override OsuSpriteText CreateSpriteText()
+            => base.CreateSpriteText().With(s => s.Font = s.Font.With(size: 20f, fixedWidth: true));
+
+        protected override LocalisableString FormatCount(double count)
+        {
+            return $@"{count:0} BPM";
+        }
+
+        public bool UsesFixedAnchor { get; set; }
+    }
+}
```

Comment:
Since this is now a counter on its own, you may want to create your own `TextComponent` rather than just putting the unit directly in `FormatCount`'s return value to match other counters.
Take a look at how `PerformancePointsCounter` and `UnstableRateCounter` do it (at this point you can just copy and paste):

https://github.com/ppy/osu/blob/01ceaa08e8bb9c03949d4880b8f5bfed3e76345a/osu.Game/Screens/Play/HUD/UnstableRateCounter.cs#L71-L74

https://github.com/ppy/osu/blob/01ceaa08e8bb9c03949d4880b8f5bfed3e76345a/osu.Game/Screens/Play/HUD/UnstableRateCounter.cs#L86-L123

Reply:
Thats an arbitrary choice, i checked the default combo counter which does it as i do and is also a standalone counter implementation

Reply:
Yes but that's because the *"unit"* is just one character long and I guess it's ok for it to be displayed in the same string (same thing with accuracy). 
Whereas units like *PP*, *UR* and *BPM* are units that take a lot of place in the playfield and should not stand out as much as the actual value, hence why the unit text was reduced (I guess).

In my opinion, having the "BPM" text reduced looks better (and keeps consistency with other counter as well)

Reply:
Was misunderstanding the complaint, will adress